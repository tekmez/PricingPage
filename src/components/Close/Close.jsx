import React from 'react'
import { TouchableOpacity } from 'react-native'
import CloseIcon from '../../../assets/svg/close.svg'
import styles from './Close.style'

const Close = ({ onPress }) => {
  return (
    <TouchableOpacity
      testID="close-button"
      style={styles.container}
      onPress={onPress}
    >
      <CloseIcon width={36} height={36} />
    </TouchableOpacity>
  )
}

export default Close
