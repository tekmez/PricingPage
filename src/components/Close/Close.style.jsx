import { StyleSheet, Platform } from 'react-native'
const topAlign = Platform.OS === 'ios' ? 52 : StatusBar.currentHeight
const rightAlign = Platform.OS === 'ios' ? 42 : StatusBar.currentHeight - 24
const styles = StyleSheet.create({
  container: {
    position: 'absolute',
    top: topAlign,
    right: rightAlign,
  },
})

export default styles
