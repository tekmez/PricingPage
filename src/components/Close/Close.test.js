import React from 'react'
import { render, fireEvent } from '@testing-library/react-native'
import Close from './Close'
jest.mock('../../../assets/svg/close.svg', () => () => (
  <div testID="close-icon" />
))
describe('Close component', () => {
  it('should match snapshot', () => {
    const tree = render(<Close />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should call onPress event', () => {
    const handlePress = jest.fn()
    const comp = render(<Close onPress={handlePress} />)
    const closeButton = comp.getByTestId('close-button')
    fireEvent.press(closeButton)
    expect(handlePress).toBeCalled()
  })
})
