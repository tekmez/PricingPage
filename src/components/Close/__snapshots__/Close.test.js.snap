// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Close component should match snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
      "position": "absolute",
      "right": 42,
      "top": 52,
    }
  }
  testID="close-button"
>
  <div
    testID="close-icon"
  />
</View>
`;
